.github/workflows/code-checks.yml

```diff
@@ -55,7 +55,7 @@ jobs:
       strategy:
         fail-fast: false
         matrix:
-          python-version: ["3.12"]
+          python-version: ["3.9"]
```

Comment:
why this change?

Reply:
Felt safer to run on the lowest version we support to make sure we don't introduce any functionality that only works on >3.9.